relayer/relays/execution/main.go

```diff
@@ -240,9 +272,10 @@ func (r *Relay) findEventsWithFilter(opts *bind.FilterOpts, paraID uint32, start
 			}
 			break
 		}
-
-		events = append(events, iter.Event)
-		if iter.Event.Nonce == start {
+		if iter.Event.Nonce >= start {
+			events = append(events, iter.Event)
+		}
+		if iter.Event.Nonce == start && opts.Start != 0 {
```

Comment:
For `opts.Start == 0` is equivalent to the case `blockNumber < BlocksPerQuery` which is the first round of positive search, not stop at the first matching and just continue until all matching events found.

Reply:
Thanks for clarifying! :)